server/src/main/java/org/apache/iotdb/db/query/executor/LastQueryExecutor.java

```diff
@@ -215,26 +220,35 @@ public QueryDataSet execute(QueryContext context, LastQueryPlan lastQueryPlan)
 
     // Compute Last result for the rest series paths by scanning Tsfiles
     int index = 0;
-    for (int i = 0; i < resultContainer.size(); i++) {
-      if (Boolean.FALSE.equals(resultContainer.get(i).left)) {
-        resultContainer.get(i).right = readerList.get(index++).readLastPoint();
-        if (resultContainer.get(i).right.getValue() != null) {
-          resultContainer.get(i).left = true;
-          if (CACHE_ENABLED) {
-            cacheAccessors.get(i).write(resultContainer.get(i).right);
-            if (context.isDebug()) {
-              DEBUG_LOGGER.info(
-                  "[LastQueryExecutor] Update last cache for path: {} with timestamp: {}",
-                  seriesPaths,
-                  resultContainer.get(i).right.getTimestamp());
-            }
+    for (int i = 0; i < cachedLastPairs.size(); i++) {
+      if (cachedLastPairs.get(i).right == null) {
```

Comment:
Here, if the filter is gt (greater than) or ge (greater than or equal to) and cachedLastPairs.get(i).left is Boolean.TRUE, we don't need to read from disk again.

Reply:
Good catch! I am busy working on a test right now, will update the PR in 2 days.